nova_plugin/server.py

```diff
@@ -287,16 +287,11 @@ def rename(name):
     ctx.logger.debug(
         "server.create() server before transformations: {0}".format(server))
 
-    for key in 'block_device_mapping', 'block_device_mapping_v2':
-        if key in server:
-            # if there is a connected boot volume, don't require the `image`
-            # property.
-            # However, python-novaclient requires an `image` input anyway, and
-            # checks it for truthiness when deciding whether to pass it along
-            # to the API
-            if 'image' not in server:
-                server['image'] = ctx.node.properties.get('image')
-            break
+    if any(['block_device_mapping' in server.keys(),
```

Comment:
note that the `any()` here is just an odd way of writing `or`.. :)
additionally, the `.keys()` isn't needed, simply `in server` will work (the `'image'` part even does that, omitting the `.keys()`!)

(BTW, in python 2, `dict.keys()` returns a list, in which `in` checking is O(N) as opposed to the O(1) `in` checking in dicts - of course in this case this doesn't matter, because server is a very small dict)

Reply:
Thanks for the review! I have made some changes. Let me know how they work for you.

Reply:
nice!